File: Mona-server/Model/Dto/MessageRequest.cs
namespace Mona.Model.Dto;

public struct MessageRequest
{
    public string? Id { get; set; }
    public string? Text { get; set; }
    public string ReceiverId { get; set; }
    public string? ChatId { get; set; }
    public string? ReplyId { get; set; }
    public string? ForwardId { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: Mona_desktop/lib/core/dto/message_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'message_request.g.dart';

@JsonSerializable()
class MessageRequest {
  final String? text;
  final String receiverId;
  final String? chatId;
  final String? replyId;
  final String? forwardId;
  final DateTime createdAt;

  MessageRequest(
      {required this.text,
      required this.receiverId,
      required this.chatId,
      required this.replyId,
      required this.forwardId,
      required this.createdAt});

  factory MessageRequest.fromJson(Map<String, dynamic> json) =>
      _$MessageRequestFromJson(json);

  Map<String, dynamic> toJson() => _$MessageRequestToJson(this);
}
